src/Transport/AzureServiceBusTransport.cs

```diff
@@ -17,20 +17,32 @@ public class AzureServiceBusTransport : TransportDefinition
         /// <summary>
         /// Creates a new instance of <see cref="AzureServiceBusTransport"/>.
         /// </summary>
-        public AzureServiceBusTransport(string connectionString) : base(
+        public AzureServiceBusTransport(string connectionString) : this()
+        {
+            Guard.AgainstNullAndEmpty(nameof(connectionString), connectionString);
+            ConnectionString = connectionString;
+        }
+
+        // Used by the shim API to enable easier migration from the existing API
+        internal AzureServiceBusTransport() : base(
```

Comment:
Nitpicking: the class can be made partial and this ctor can be moved to the `obsoletes` file so that all stuff to be removed is in a single place.

Reply:
done